### PR TITLE
add jackson serialization

### PR DESCRIPTION
**Pull request by jsbxyyx**
fix: #1116

### CODE REVIEW

**Comment by zhangthen** [comment]
There was a PR (#1125 )  about  adding protostuff as the serializer of UndoLogParser. JSON is inefficient, is it necessary to add a jackson serialization ?

File: rm-datasource/src/test/java/io/seata/rm/datasource/undo/BaseH2Test.java [hunk 1]
```diff
@@ -15,23 +15,14 @@
...
 import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.BeforeEach;
 import org.mockito.Mockito;
 
-import java.sql.Connection;
-import java.sql.ResultSet;
-import java.sql.SQLException;
-import java.sql.Statement;
-import java.sql.Types;
+import java.sql.*;
```
Comment (slievrly) [inline rm-datasource/src/test/java/io/seata/rm/datasource/undo/BaseH2Test.java]: forbid to use *

File: rm-datasource/src/main/java/io/seata/rm/datasource/DataCompareUtils.java [hunk 2]
```diff
@@ -57,14 +55,6 @@ public static boolean isFieldEquals(Field f0, Field f1) {
                         if (f1.getValue() == null) {
                             return false;
                         } else {
-                            int f0Type = f0.getType();
```
Comment (slievrly) [inline rm-datasource/src/main/java/io/seata/rm/datasource/DataCompareUtils.java]: I want to temporarily keep fastjson, can you make a judgment?

Comment (jsbxyyx) [reply rm-datasource/src/main/java/io/seata/rm/datasource/DataCompareUtils.java]: yes

**Comment by slievrly** [comment]
Functional Verification: bigint(11), date, time, timestamp check ok.
CREATE TABLE `test_jackson` (
  `id` int(11) NOT NULL AUTO_INCREMENT,
  `int_test` bigint(11) DEFAULT NULL,
  `date_test` date DEFAULT NULL,
  `time_test` time DEFAULT NULL,
  `timespan_test` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB AUTO_INCREMENT=4 DEFAULT CHARSET=utf8

File: rm-datasource/src/main/java/io/seata/rm/datasource/undo/parser/JacksonUndoLogParser.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+    static {
+        MAPPER.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+        MAPPER.enableDefaultTyping(ObjectMapper.DefaultTyping.NON_FINAL, JsonTypeInfo.As.PROPERTY);
+    }
+
+    @Override
+    public String getName() {
+        return "jackson";
```
Comment (slievrly) [inline rm-datasource/src/main/java/io/seata/rm/datasource/undo/parser/JacksonUndoLogParser.java]: Constant as fastjson

File: rm-datasource/src/main/java/io/seata/rm/datasource/undo/UndoLogManager.java [hunk 4]
```diff
@@ -187,6 +192,8 @@ public static void undo(DataSourceProxy dataSourceProxy, String xid, long branch
                             sqlUndoLog);
                         undoExecutor.executeOn(conn);
                     }
+                    // remove serializer name
+                    SERIALIZER_LOCAL.remove();
```
Comment (slievrly) [inline rm-datasource/src/main/java/io/seata/rm/datasource/undo/UndoLogManager.java]: better add in finally block？

File: rm-datasource/src/main/java/io/seata/rm/datasource/undo/parser/FastjsonUndoLogParser.java [hunk 5]
```diff
@@ -27,12 +27,15 @@
  *
  * @author sharajava
  */
-@LoadLevel(name = "fastjson")
+@Deprecated
```
Comment (CoffeeLatte007) [inline rm-datasource/src/main/java/io/seata/rm/datasource/undo/parser/FastjsonUndoLogParser.java]: why Deprecated ?

**Pull request merged**